### PR TITLE
Hello from Catlin - Catalog Linter 😻

### PR DESCRIPTION
**Pull request by sthaha**
# Changes
Catlin is a command-line Tekton resource (yaml) linter that validates a
catalog or an individual resource against the guidelines specified in the
Catalog Organization TEP - https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md

Catlin current support only `validate` sub-command that validates a
single yaml file.

Usage: `catlin validate <file>`

Signed-off-by: Sunil Thaha <sthaha@redhat.com

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [🙅 ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: catlin/pkg/validator/file_validator.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		result.Error("invalid path: %s - %s", v.path, err)
+		return result
+	}
+
+	expectedPath := filepath.Join(kind, name, version, name+".yaml")
+
+	//v.log.Infof("name: %s | kind: %s | version: %s", name, kind, version)
+	//v.log.Infof("abspath: %s", absPath)
+	//v.log.Infof("expectedPath: %s", expectedPath)
```
Comment (savitaashture) [inline catlin/pkg/validator/file_validator.go:56]: nit

can we remove these lines if it added for debug purpose

File: catlin/pkg/validator/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	r := strings.NewReader(validTask)
+	parser := parser.ForReader(r)
+
+	res, err := parser.Parse()
+	assert.NilError(t, err)
+
+	v := NewContentValidator(tc, res)
+	result := v.Validate()
+
+	//t.Logf("%v", result.Lints)
```
Comment (savitaashture) [inline catlin/pkg/validator/validate_test.go:103]: same here

File: catlin/pkg/consts/consts.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consts
+
+const (
+	VersionLabel                 = "app.kubernetes.io/version"
+	DisplayNameAnnotation        = "tekton.dev/displayName"
+	MinPipelineVersionAnnotation = "tekton.dev/pipelines.minVersion"
+	TagsAnnotation               = "tekton.dev/tags"
```
Comment (savitaashture) [inline catlin/pkg/consts/consts.go:21]: I see right now Catlin is checking these labels/annotation 
But `Catlin` don't have information about optional and mandatory labels/annotation

Just a suggestion like can we add doc somewhere to give just a brief about the `Catlin` and what all things it validates something like that
But ya if you planned documentation sometime later as part of another PR that's totally fine

**Review by savitaashture — COMMENTED** [review]
PR looks clean and structured way  
Thank you :slightly_smiling_face: @sthaha :bowing_woman: 

Just few minor points 
otherwise 

/lgtm :+1:

File: catlin/pkg/consts/consts.go (hunk 3 above)
Comment (sthaha) [reply catlin/pkg/consts/consts.go:21]: @savitaashture these labels and annotations are documented in the TEP  - https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md 
which is what it tries to validate against.

Comment (afrittoli) [reply catlin/pkg/consts/consts.go:21]: +1 on adding docs, it's fine to do so in a follow-up PR, but it would be nice to have at least a basic readme.

**Review by afrittoli — COMMENTED** [review]
Cool, thanks for this!: :)

A few things that would be good to have:
- some basic doc on purpose / how to use
- some more unit tests and a job to run them

I wonder if this eventually could live in the catalog repo or in it's own, but it's fine to have it in plumbing, at least for. now.

/approve

**Comment by sthaha** [comment]
> Cool, thanks for this!: :)
> 
> A few things that would be good to have:
> 
> * some basic doc on purpose / how to use
> * some more unit tests and a job to run them

Thanks adding bugs now.


> 
> I wonder if this eventually could live in the catalog repo or in it's own, but it's fine to have it in plumbing, at least for. now.
> 

I agree, once the tool proves to be valuable, we should think of a better 🏠  for Catlin :)

**Pull request merged**